packages/block-library/src/navigation/edit.js

```diff
@@ -107,6 +122,47 @@ function Navigation( {
 		);
 	}, [ pages ] );
 
+	const menuItems = getMenuItems( selectedDropDownOption?.key );
+
+	// Builds navigation links from selected Menu's items.
+	const navLinkBlocksFromMenuItems = useMemo( () => {
+		if ( ! menuItems ) {
+			return null;
+		}
+
+		function initialiseBlocks( nodes ) {
+			return nodes.map( ( { title, type, link: url, id, children } ) => {
+				const innerBlocks =
+					children && children.length
+						? initialiseBlocks( children )
+						: [];
+
+				return createBlock(
+					'core/navigation-link',
+					{
+						type,
+						id,
+						url,
+						label: ! title.rendered
+							? __( '(no title)' )
+							: escape( title.rendered ),
+						opensInNewTab: false,
+					},
+					innerBlocks
+				);
+			} );
+		}
+
+		const menuTree = createDataTree( menuItems );
+
+		const menuBlocksTree = initialiseBlocks( menuTree );
+
+		return menuBlocksTree;
+	}, [ menuItems ] );
```

Comment:
Something for a follow-up: this is feeling incredibly similar to what the navigation menus page does, so I'm wondering if there's an opportunity to share some well unit-tested code (not sure there are unit tests for the nav menus page yet, but it's on the roadmap).

Reply:
Seems sensible. It needs unit testing.

I'm not super familiar with the Navigation Menus page at the moment. Would someone who's worked on this be able to create an issue informed by their knowledge of where the code has similarities?

Reply:
Yep, there are a few who've worked on it, though I haven't looked at it in a while. Lets not tackle it in this PR.